linux/jdk/suse/src/main/packaging/temurin/21/temurin-21-jdk.spec

```diff
@@ -0,0 +1,219 @@
+# https://docs.fedoraproject.org/en-US/packaging-guidelines/Versioning/#_upstream_uses_invalid_characters_in_the_version
+# also not very intuitive:
+#  $ rpmdev-vercmp 21.0.0.0.0___21.0.0.0.0+1
+#  21.0.0.0.0___1 == 21.0.0.0.0+35
+%global spec_version 21.0.0.0.0.35
+%global spec_release 1
+%global priority 1161
+
+%global source_url_base https://github.com/adoptium/temurin21-binaries/releases/download
+%global upstream_version_url %(echo %{upstream_version} | sed 's/\+/%%2B/g')
+%global upstream_version_no_plus %(echo %{upstream_version} | sed 's/\+/_/g')
+%global java_provides openjdk
+
+# Map architecture to the expected value in the download URL; Allow for a
+# pre-defined value of vers_arch and use that if it's defined
+
+%ifarch x86_64
+%global vers_arch x64
+%global vers_arch2 ppc64le
+%global vers_arch3 aarch64
+%global src_num 0
+%global sha_src_num 1
+%endif
+%ifarch ppc64le
+%global vers_arch x64
+%global vers_arch2 ppc64le
+%global vers_arch3 aarch64
+%global src_num 2
+%global sha_src_num 3
+%endif
+%ifarch aarch64
+%global vers_arch x64
+%global vers_arch2 ppc64le
+%global vers_arch3 aarch64
+%global src_num 4
+%global sha_src_num 5
+%endif
+# Allow for noarch SRPM build
+%ifarch noarch
+%global src_num 0
+%global sha_src_num 1
+%endif
+
+Name:        temurin-21-jdk
+Version:     %{spec_version}
+Release:     %{spec_release}
+Summary:     Eclipse Temurin 21 JDK
+
+Group:       java
+License:     GPLv2 with exceptions
+Vendor:      Eclipse Adoptium
+URL:         https://projects.eclipse.org/projects/adoptium
+Packager:    Eclipse Adoptium Package Maintainers <temurin-dev@eclipse.org>
+
+AutoReqProv: no
+Prefix: %{_libdir}/jvm/%{name}
+
+ExclusiveArch: x86_64 aarch64
+
+BuildRequires:  tar
+BuildRequires:  wget
+
+Requires: /bin/sh
+Requires: /usr/sbin/alternatives
+Requires: ca-certificates
+Requires: dejavu-fonts
+Requires: libX11-6%{?_isa}
+Requires: libXext6%{?_isa}
+Requires: libXi6%{?_isa}
+Requires: libXrender1%{?_isa}
+Requires: libXtst6%{?_isa}
+Requires: libasound2%{?_isa}
+Requires: glibc%{?_isa}
+Requires: libz1%{?_isa}
+Requires: fontconfig%{?_isa}
+Requires: libfreetype6%{?_isa}
+
+Provides: java
+Provides: java-21
+Provides: java-21-devel
+Provides: java-21-headless
+Provides: java-21-%{java_provides}
+Provides: java-21-%{java_provides}-devel
+Provides: java-21-%{java_provides}-headless
+Provides: java-devel
+Provides: java-devel-%{java_provides}
+Provides: java-headless
+Provides: java-%{java_provides}
+Provides: java-%{java_provides}-devel
+Provides: java-%{java_provides}-headless
+Provides: java-sdk
+Provides: java-sdk-21
+Provides: java-sdk-21-%{java_provides}
+Provides: java-sdk-%{java_provides}
+
+# First architecture (x86_64)
+Source0: %{source_url_base}/jdk-%{upstream_version_url}/OpenJDK21U-jdk_%{vers_arch}_linux_hotspot_%{upstream_version_no_plus}.tar.gz
+Source1: %{source_url_base}/jdk-%{upstream_version_url}/OpenJDK21U-jdk_%{vers_arch}_linux_hotspot_%{upstream_version_no_plus}.tar.gz.sha256.txt
```

Comment:
"U"

Also, I'm going to stop tagging all the "U"s. There's a bunch more of them, and the million one-letter-comments doesn't add value.